### PR TITLE
Fix/packages

### PR DESCRIPTION
**Pull request by pciavald**
There's an issue on the dev branch. **Cloning from scratch** results in crashing build. This PR fixes the package dependencies that origin this behaviour (the bug is known).

### fixed build error
```
Date: 2017-12-20T02:37:03.985Z                                                        
Hash: 8430f1a858b696366dc6
Time: 9918ms
chunk {inline} inline.bundle.js (inline) 5.79 kB [entry] [rendered]
chunk {main} main.bundle.js (main) 263 bytes [initial] [rendered]
chunk {polyfills} polyfills.bundle.js (polyfills) 278 bytes [initial] [rendered]
chunk {styles} styles.bundle.js (styles) 157 kB [initial] [rendered]

ERROR in node_modules/@angular/flex-layout/media-query/breakpoints/break-points.d.ts(8,10): error TS2305: Module '"/home/pciavald/Documents/partgui-kiriti/node_modules/@angular/core/core"' has no exported member 'OpaqueToken'.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! particl@1.1.0 build:electron:fast: `ng build --base-href=./ --sourcemap=false`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the particl@1.1.0 build:electron:fast script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pciavald/.npm/_logs/2017-12-20T02_37_04_012Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! particl@1.1.0 start:electron:fast: `npm run build:electron:fast && electron . "-printtoconsole"`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the particl@1.1.0 start:electron:fast script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pciavald/.npm/_logs/2017-12-20T02_37_04_117Z-debug.log
```

### However, now the GUI starts but crashes in the dev tools log : 

```
Uncaught Error: Template parse errors:
'mat-spinner' is not a known element:
1. If 'mat-spinner' is an Angular component, then verify that it is part of this module.
2. If 'mat-spinner' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message. ("

<div mat-dialog-content class="mat-wallet-content">
  [ERROR ->]<mat-spinner mode="indeterminate" *ngIf="loadSpinner"></mat-spinner>

  <ng-template #modalContainer>"): ng:///ModalsModule/ModalsComponent.html@5:2
    at syntaxError (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:63576:34)
    at TemplateParser.webpackJsonp.../../../compiler/esm5/compiler.js.TemplateParser.parse (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:87722:19)
    at JitCompiler.webpackJsonp.../../../compiler/esm5/compiler.js.JitCompiler._parseTemplate (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97533:37)
    at JitCompiler.webpackJsonp.../../../compiler/esm5/compiler.js.JitCompiler._compileTemplate (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97508:23)
    at file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97409:62
    at Set.forEach (native)
    at JitCompiler.webpackJsonp.../../../compiler/esm5/compiler.js.JitCompiler._compileComponents (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97409:19)
    at file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97279:19
    at Object.then (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:63565:77)
    at JitCompiler.webpackJsonp.../../../compiler/esm5/compiler.js.JitCompiler._compileModuleAndComponents (file:///home/pciavald/Documents/partgui-clean/dist/vendor.bundle.js:97278:26)
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -93,20 +93,19 @@
...
-    "@angular/platform-browser": "^5.0.3",
-    "@angular/platform-browser-dynamic": "^5.0.3",
-    "@angular/platform-server": "^5.0.3",
-    "@angular/router": "^5.0.3",
+    "@angular/animations": "^5.1.1",
+    "@angular/cdk": "^5.0.2",
+    "@angular/common": "^5.1.1",
+    "@angular/compiler": "^5.1.1",
+    "@angular/core": "^5.1.1",
+    "@angular/flex-layout": "^2.0.0-beta.12",
```
Comment (pciavald) [inline package.json]: https://github.com/angular/flex-layout-builds/blob/master/CHANGELOG.md#200-beta12-2017-12-14

File: package.json [hunk 2]
```diff
@@ -93,20 +93,19 @@
...
-    "@angular/material": "2.0.0-beta.12",
-    "@angular/platform-browser": "^5.0.3",
-    "@angular/platform-browser-dynamic": "^5.0.3",
-    "@angular/platform-server": "^5.0.3",
-    "@angular/router": "^5.0.3",
+    "@angular/animations": "^5.1.1",
+    "@angular/cdk": "^5.0.2",
+    "@angular/common": "^5.1.1",
+    "@angular/compiler": "^5.1.1",
+    "@angular/core": "^5.1.1",
```
Comment (pciavald) [inline package.json]: https://blog.angular.io/angular-5-1-more-now-available-27d372f5eb4e

File: package.json [hunk 3]
```diff
@@ -124,13 +123,13 @@
     "node-unzip-2": "^0.2.1",
     "open-sans-all": "^0.1.3",
     "rx-ipc-electron": "^0.1.2",
-    "rxjs": "5.5.0",
+    "rxjs": "5.5.2",
```
Comment (pciavald) [inline package.json]: https://github.com/angular/flex-layout/issues/483#issuecomment-341940553

**Comment by pciavald** [comment]
@cryptoguard can you tell me what that gives on your side ?

**Comment by pciavald** [comment]
@HenkSwardt when the material issue of this branch is fixed, we can close **without merging** all linked PRs.

File: package.json [hunk 4]
```diff
@@ -153,6 +152,6 @@
     "ts-node": "~3.3.0",
     "tslint": "~5.8.0",
     "tslint-microsoft-contrib": "^5.0.1",
-    "typescript": "^2.6.1"
+    "typescript": "2.4.2"
```
Comment (pciavald) [inline package.json]:
```
@angular/compiler-cli@5.0.3 requires typescript@'>=2.4.2 <2.5.0' but 2.6.2 was found instead.
Using this version can result in undefined behaviour and difficult to debug problems.

Please run the following command to install a compatible version of TypeScript.

    npm install typescript@'>=2.4.2 <2.5.0'

To disable this warning run "ng set warnings.typescriptMismatch=false".
```

**Comment by pciavald** [comment]
# update

Errors were fixed and syntaxes adapted to the new `pipe(map())` rxjs operator, but now this error comes up in the console : 

```
ERROR TypeError: __WEBPACK_IMPORTED_MODULE_1_rxjs_Observable__.a.create(...).shareReplay is not a function
    at StateService.webpackJsonp.../../../../../src/app/core/state/state.service.ts.StateService._getObservablePair (main.bundle.js:1672)
    at StateService.webpackJsonp.../../../../../src/app/core/state/state.service.ts.StateService.observe (main.bundle.js:1659)
    at RpcStateClass.webpackJsonp.../../../../../src/app/core/rpc/rpc-state/rpc-state.class.ts.RpcStateClass.lastBlockTimeState (main.bundle.js:1228)
    at new RpcStateClass (main.bundle.js:1219)
    at RpcService.webpackJsonp.../../../../../src/app/core/rpc/rpc.service.ts.RpcService.toggleState (main.bundle.js:1553)
    at new RpcService (main.bundle.js:1422)
    at _createClass (vendor.bundle.js:117542)
    at _createProviderInstance$1 (vendor.bundle.js:117512)
    at resolveNgModuleDep (vendor.bundle.js:117497)
    at NgModuleRef_.webpackJsonp.../../../core/esm5/core.js.NgModuleRef_.get (vendor.bundle.js:118733)
```

@rynomster we need your help on this. The new version of rxjs required by the new version of angular required by the bugged and updated version of flex-layout is the source of the problem. 

HttpModule was deprecated and replaced by HttpClient, the interface of http changed and is already adapted in this PR. RxJS probably updated how their Observables work which leads to this error ^.

The error traces back to that line where shareReplay() is not a function on these new Observables.
https://github.com/particl/partgui/blob/4b3a86565297c7ef132f5619b442162c8fb2671b/src/app/core/state/state.service.ts#L73

File: package.json (hunk 3 above)
Comment (pciavald) [reply package.json]: https://github.com/redux-observable/redux-observable/issues/341

**Comment by pciavald** [comment]
@cryptoguard should be good this time :)

**Comment by allienworks** [comment]
Works for me ✔

**Pull request merged**